commands/generate-messages.ts

```diff
@@ -0,0 +1,24 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import * as path from "path";
+
+export class GenerateMessages implements ICommand {
+	constructor(private $fs: IFileSystem,
+				private $messageContractGenerator: IServiceContractGenerator,
+				private $staticConfig: Config.IStaticConfig) {
+	}
+
+	allowedParameters: ICommandParameter[] = [];
+
+	execute(args: string[]): IFuture<void> {
+		return (() => {
+			let result = this.$messageContractGenerator.generate().wait(),
+				interfaceFilePath = this.$staticConfig.CLIENT_NAME ? path.join(__dirname, "../../messages.d.ts") : path.join(__dirname, "../messages/messages.d.ts"),
+				implementationFilePath = this.$staticConfig.CLIENT_NAME ? path.join(__dirname, "../../messages.ts") : path.join(__dirname, "../messages/messages.ts");
```

Comment:
You can share the check between these two lines. Extract the directory path into a separate variable and then append the file name.